### PR TITLE
Customer Application 

### PR DESCRIPTION
**Pull request by emtec-damodhar-jadhao**
Customer Application Ui and Backend folders Added on 16-December

### CODE REVIEW

File: Customer_Application/CustomerEndpoint/userDataHandler.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿using NServiceBus.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UserData;
+
```
Comment (nitish-poojari) [inline Customer_Application/CustomerEndpoint/userDataHandler.cs]: Move to inside folder Handler

File: Customer_Application/CustomerEndpoint/userDataHandler.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿using NServiceBus.Logging;
```
Comment (nitish-poojari) [inline Customer_Application/CustomerEndpoint/userDataHandler.cs]: move inside namespace

File: Customer_Application/Customer_Api/Customer_Api.csproj [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <ImplicitUsings>enable</ImplicitUsings>
+  </PropertyGroup>
+
```
Comment (nitish-poojari) [inline Customer_Application/Customer_Api/Customer_Api.csproj:8]: exclude using config

File: Customer_Application/Customer_Api/Controllers/CustomerDataController.cs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    [ApiController]
+    public class CustomerDataController : ControllerBase
+    {
+        private readonly IConfiguration _configuration;
+        private readonly IDataBaseOperation _DataBaseOperation;
+        private readonly ILogger<CustomerDataController> _logger;
+        private readonly IMessageSession session;
+
+
+        public CustomerDataController(IConfiguration configuration, IDataBaseOperation DataBaseOperation, ILogger<CustomerDataController> logger, IMessageSession session)
```
Comment (nitish-poojari) [inline Customer_Application/Customer_Api/Controllers/CustomerDataController.cs]: CustomerController

File: Customer_Application/Customer_Api/Controllers/CustomerDataController.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            _configuration = configuration;
+            _DataBaseOperation = DataBaseOperation;
+            _logger = logger;
+            this.session = session;
+        }
+
+        [HttpGet ("getallcustomer")]
+        public async Task<ActionResult<List<object>>> GetAllCustomerData()
+        {
+            using var connection = new SqlConnection(_configuration.GetConnectionString("DefaultConnection"));
```
Comment (nitish-poojari) [inline Customer_Application/Customer_Api/Controllers/CustomerDataController.cs]: move to infrastructure

File: Customer_Application/Customer_Api/Controllers/CustomerDataController.cs [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            var Customers = await connection.QueryAsync<object>(_DataBaseOperation.GetAllCustomerData());
+            return Ok(Customers);
+        }
+        [HttpGet("getbyid")]
+        public async Task<ActionResult<CustomerData>> getcustomerbyid(int customerid)
+        {
+            using var connection = new SqlConnection(_configuration.GetConnectionString("DefaultConnection"));
```
Comment (nitish-poojari) [inline Customer_Application/Customer_Api/Controllers/CustomerDataController.cs]: Should not be creating connection every time when endpoint is called. Move All DB operations to seperate infrastructure project and use dependancy injection here

File: Customer_Application/Customer_Api/Infrastructure/DataBaseOperation.cs [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        public string GetAllCustomerData();
+        public string getbyid(int id);
+        public string getbyname(string customername);
+        public string addcustomer(CustomerData newcustomer);
+        public string updatecustomer(CustomerData newcustomer);
+        public string deletecustomer(int id);
+         
+    }
+
+
```
Comment (nitish-poojari) [inline Customer_Application/Customer_Api/Infrastructure/DataBaseOperation.cs]: Create new lib project as infrastructure and move it

File: Customer_Application/Customer_Api/Infrastructure/DataBaseOperation.cs [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        public string GetAllCustomerData();
+        public string getbyid(int id);
+        public string getbyname(string customername);
+        public string addcustomer(CustomerData newcustomer);
+        public string updatecustomer(CustomerData newcustomer);
+        public string deletecustomer(int id);
+         
+    }
+
```
Comment (nitish-poojari) [inline Customer_Application/Customer_Api/Infrastructure/DataBaseOperation.cs]: move to seperate file

File: Customer_Application/Customer_Api/Infrastructure/DataBaseOperation.cs [hunk 9]
```diff
@@ -0,0 +1,54 @@
+﻿using Customer_Api.Model;
+
```
Comment (nitish-poojari) [inline Customer_Application/Customer_Api/Infrastructure/DataBaseOperation.cs]: move inside namespace

File: Customer_Application/UserData/UserDataAdd.cs [hunk 10]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NServiceBus;
+
+namespace UserData
+{
```
Comment (nitish-poojari) [inline Customer_Application/UserData/UserDataAdd.cs]: Create seprate lib project as model and move all models thr. use the reference of model project in this project

File: Customer_Application/Customer_Api/Controllers/CustomerDataController.cs [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+namespace Customer_Api.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CustomerDataController : ControllerBase
+    {
+        private readonly IConfiguration _configuration;
+        private readonly IDataBaseOperation _DataBaseOperation;
+        private readonly ILogger<CustomerDataController> _logger;
+        private readonly IMessageSession session;
```
Comment (adi-source) [inline Customer_Application/Customer_Api/Controllers/CustomerDataController.cs]: _session

File: Customer_Application/Customer_Api/Controllers/CustomerDataController.cs [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+
+
+        public CustomerDataController(IConfiguration configuration, IDataBaseOperation DataBaseOperation, ILogger<CustomerDataController> logger, IMessageSession session)
+        {
+            _configuration = configuration;
+            _DataBaseOperation = DataBaseOperation;
+            _logger = logger;
+            this.session = session;
```
Comment (adi-source) [inline Customer_Application/Customer_Api/Controllers/CustomerDataController.cs]: _session

File: Customer_Application/Customer_Api/Infrastructure/DataBaseOperation.cs [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+namespace Customer_Api.Infrastructure
+{
+    public interface IDataBaseOperation
+    {
+        public string GetAllCustomerData();
+        public string getbyid(int id);
+        public string getbyname(string customername);
+        public string addcustomer(CustomerData newcustomer);
+        public string updatecustomer(CustomerData newcustomer);
+        public string deletecustomer(int id);
```
Comment (adi-source) [inline Customer_Application/Customer_Api/Infrastructure/DataBaseOperation.cs]: Use Camel/Pascal case for all variable declarations

File: Customer_Application/CustomerEndpoint/userDataHandler.cs (hunk 1 above)
Comment (emtec-damodhar-jadhao) [reply Customer_Application/CustomerEndpoint/userDataHandler.cs]: Added Folder handler

File: Customer_Application/CustomerEndpoint/userDataHandler.cs (hunk 2 above)
Comment (emtec-damodhar-jadhao) [reply Customer_Application/CustomerEndpoint/userDataHandler.cs]: Moved Using Inside NameSpaces

File: Customer_Application/Customer_Api/Controllers/CustomerDataController.cs (hunk 4 above)
Comment (emtec-damodhar-jadhao) [reply Customer_Application/Customer_Api/Controllers/CustomerDataController.cs]: Change controller name To CustomerController

File: Customer_Application/Customer_Api/Controllers/CustomerDataController.cs (hunk 5 above)
Comment (emtec-damodhar-jadhao) [reply Customer_Application/Customer_Api/Controllers/CustomerDataController.cs]: Connection String moved in Infrastructure

File: Customer_Application/Customer_Api/Controllers/CustomerDataController.cs (hunk 6 above)
Comment (emtec-damodhar-jadhao) [reply Customer_Application/Customer_Api/Controllers/CustomerDataController.cs]: dependency Injection Used

File: Customer_Application/Customer_Api/Infrastructure/DataBaseOperation.cs (hunk 7 above)
Comment (emtec-damodhar-jadhao) [reply Customer_Application/Customer_Api/Infrastructure/DataBaseOperation.cs]: database opearation added in class library project Infrastructure

File: Customer_Application/Customer_Api/Infrastructure/DataBaseOperation.cs (hunk 8 above)
Comment (emtec-damodhar-jadhao) [reply Customer_Application/Customer_Api/Infrastructure/DataBaseOperation.cs]: Moved in infrastructure

File: Customer_Application/Customer_Api/Infrastructure/DataBaseOperation.cs (hunk 9 above)
Comment (emtec-damodhar-jadhao) [reply Customer_Application/Customer_Api/Infrastructure/DataBaseOperation.cs]: Using Added inside Namespaces

File: Customer_Application/UserData/UserDataAdd.cs (hunk 10 above)
Comment (emtec-damodhar-jadhao) [reply Customer_Application/UserData/UserDataAdd.cs]: Created lib project as model and used reference in this project

File: Customer_Application/Customer_Api/Controllers/CustomerDataController.cs (hunk 11 above)
Comment (emtec-damodhar-jadhao) [reply Customer_Application/Customer_Api/Controllers/CustomerDataController.cs]: take instance as _session

File: Customer_Application/Customer_Api/Controllers/CustomerDataController.cs (hunk 12 above)
Comment (emtec-damodhar-jadhao) [reply Customer_Application/Customer_Api/Controllers/CustomerDataController.cs]: made change

File: Customer_Application/Customer_Api/Infrastructure/DataBaseOperation.cs (hunk 13 above)
Comment (emtec-damodhar-jadhao) [reply Customer_Application/Customer_Api/Infrastructure/DataBaseOperation.cs]: Used Camel casing for variable declaration

**Pull request merged**